ivy/functional/backends/paddle/activations.py

```diff
@@ -13,33 +13,23 @@
 # local
 import ivy.functional.backends.paddle as paddle_backend
 import ivy
-from ivy.func_wrapper import with_unsupported_device_and_dtypes
+from ivy.func_wrapper import with_unsupported_device_and_dtypes, with_supported_dtypes
 from . import backend_version
 
 
-unsupported_dtypes = [
-    paddle.int8,
-    paddle.int16,
-    paddle.int32,
-    paddle.int64,
-    paddle.uint8,
-    paddle.float16,
-    paddle.complex64,
-    paddle.complex128,
-    paddle.bool,
-]
-
-
+@with_supported_dtypes(
+    {"2.5.1 and below": ("float16", "uint16", "float32", "float64", "complex")},
+    backend_version,
+)
 def relu(x: paddle.Tensor, /, *, out: Optional[paddle.Tensor] = None) -> paddle.Tensor:
-    if x.dtype in unsupported_dtypes:
-        if paddle.is_complex(x):
-            return paddle.complex(F.relu(x.real()), F.relu(x.imag()))
-        return F.relu(x.cast("float32")).cast(x.dtype)
+    if paddle.is_complex(x):
+        return paddle.complex(F.relu(x.real()), F.relu(x.imag()))
     return F.relu(x)
 
 
-@with_unsupported_device_and_dtypes(
-    {"2.5.1 and below": {"cpu": ("bfloat16",)}}, backend_version
+@with_supported_dtypes(
+    {"2.5.1 and below": ("float16", "uint16", "float32", "float64", "complex")},
```

Comment:
instead of `float16`, `float32` and `float64` you could just use `float`, as all float dtypes are supported.
Also, I think [`uint16`](https://github.com/unifyai/ivy/blob/47359a574231bd476f54fea63ee4fd8105d68991/ivy/functional/backends/paddle/__init__.py#L139) isn't supported in `paddle` 
Same applies to other similar instances in the `paddle` backend

Reply:
Actually I looked into the functions source code to figure out the supported types, for example: [here](https://github.com/PaddlePaddle/Paddle/blob/daac3829d87d2a96b6015c946bcf6fa82a690fd2/python/paddle/nn/functional/activation.py#L458) I think is where the supported types checks if performed and uint16 is mentioned.

Reply:
Yeah it seems like they might've started supporting `uint16` in the later versions, @MahmoudAshraf97 what do you think? Thanks!